### ISSUE TITLE
Child sets within parent set with percave="true" are weird

### ISSUE THREAD

**Issue by A2ur3** [FakeFishGames/Barotrauma#13316]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

```
    <EventSet identifier="caves" chooserandom="true" percave="true" triggereventcooldown="false" allowatstart="true">
      <EventSet chooserandom="false" commonness="75" allowatstart="true">
        <MonsterEvent characterfile="Crawler" minamount="4" maxamount="5" spawntype="cave" />
        <MonsterEvent characterfile="Crawlerbroodmother" minamount="1" maxamount="1" spawntype="cave" />
      </EventSet>
      <EventSet chooserandom="false" commonness="20" allowatstart="true">
        <MonsterEvent characterfile="Mudraptor" minamount="3" maxamount="4" spawntype="cave" />
        <MonsterEvent characterfile="Mudraptor_veteran" minamount="2" maxamount="3" spawntype="cave" />
      </EventSet>
      <EventSet chooserandom="false" commonness="5" allowatstart="true">
        <MonsterEvent characterfile="Tigerthresher" minamount="3" maxamount="4" spawntype="cave" />
        <MonsterEvent characterfile="Bonethresher" minamount="1" maxamount="2" spawntype="cave" />
      </EventSet>
    </EventSet>
```

I'm using the Eventset above.
from my understanding, this Eventset should be applied to each cave in the level
and 75% cave is crawler spawn, 20% is mudraptor spawn, 5% is tigerthresher spawn
but the fact is all child sets are chosen and applied to one cave, making numerous monster spawn in one cave and no monster spawn in other caves

![eventset](https://github.com/Regalis11/Barotrauma/assets/64390906/ef698d48-c635-43ba-9707-45ce882b25a9)

### Reproduction steps

include child sets within percave="true" parent set

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I would argue that this is working correctly. 

The parent set is executed once per cave, but the child sets are not, and there's nothing in them telling the game they should take place in different caves. I think you should remove `percave="true"` from the parent set and move it to the child sets instead.

**Comment by A2ur3** [comment]
thanks for your answer. I did what you suggested.
`<EventSet identifier="caves" chooserandom="true" percave="false" triggereventcooldown="false" allowatstart="true">
  <EventSet chooserandom="false" commonness="75" allowatstart="true" percave="true">
    <MonsterEvent characterfile="Crawler" minamount="4" maxamount="5" spawntype="cave" />
    <MonsterEvent characterfile="Crawlerbroodmother" minamount="1" maxamount="1" spawntype="cave" />
  </EventSet>
  <EventSet chooserandom="false" commonness="20" allowatstart="true" percave="true">
    <MonsterEvent characterfile="Mudraptor" minamount="3" maxamount="4" spawntype="cave" />
    <MonsterEvent characterfile="Mudraptor_veteran" minamount="2" maxamount="3" spawntype="cave" />
  </EventSet>
  <EventSet chooserandom="false" commonness="5" allowatstart="true" percave="true">
    <MonsterEvent characterfile="Tigerthresher" minamount="3" maxamount="4" spawntype="cave" />
    <MonsterEvent characterfile="Bonethresher" minamount="1" maxamount="2" spawntype="cave" />
  </EventSet>
</EventSet>`
But this makes same eventset applied in same cave six times
![eventset](https://github.com/Regalis11/Barotrauma/assets/64390906/b2845032-e257-4b4f-ab48-8e7a77606886)

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/99724e34f2ca5b5c0fc2e9629ad3676bade7c310

**Comment by itchyOwl** [comment]
Confirmed to be fixed in FakeFishGames/Barotrauma-development#4631.
